# Task 32 — High-Usage Customers by Selected Plans

## Business Question

How many high-usage customers are using selected subscription plans?

## Analysis

Filtered customers whose subscription plan is Business, Residential, or Roam and whose gb_downloaded exceeds 3,000 GB. The filtered customers were then grouped by subscription plan, counted, and sorted in descending order by the number of high-usage customers.

## Code

In [1]:
import pandas as pd
import numpy as np
customers = pd.read_csv("starlink_customers.csv")
subscriptions = pd.read_csv("subscriptions.csv")
starlink = customers.merge(
    subscriptions,
    left_on = "unique_id",
    right_on = "customer_id",
    how = "left"
)
starlink_users = starlink[
        starlink["plan_name"].isin(["Business", "Residential", "Roam"])
        & (starlink["gb_downloaded"] > 3000)
    ].groupby("plan_name").agg(
        users_count = ("unique_id", "count")
    ).sort_values(
        "users_count",
        ascending = False
    )
print(starlink_users)

             users_count
plan_name               
Residential          793
Roam                 285
Business             146


### Business Insight

Residential has the highest number of high-usage customers (793), followed by Roam (285) and Business (146). This indicates that Residential customers represent the largest group of high-usage customers among the selected subscription plans.